astropy/convolution/convolve.py

```diff
@@ -406,11 +416,11 @@ def convolve_fft(array, kernel, boundary='fill', fill_value=0, crop=True,
         kernel[mask] = np.nan
 
     # NaN and inf catching
-    nanmaskarray = np.isnan(array) + np.isinf(array)
+    nanmaskarray = np.isnan(array) | np.isinf(array)
     array[nanmaskarray] = 0
-    nanmaskkernel = np.isnan(kernel) + np.isinf(kernel)
+    nanmaskkernel = np.isnan(kernel) | np.isinf(kernel)
     kernel[nanmaskkernel] = 0
-    if ((nanmaskarray.sum() > 0 or nanmaskkernel.sum() > 0) and
+    if ((np.count_nonzero(nanmaskarray) > 0 or np.count_nonzero(nanmaskkernel) > 0) and
```

Comment:
To help prevent unnecessary looping over the whole mask arrays, I'd write it as
```
if (not interpolate_nan and not quite and
    (np.any(nanmasarray) or np.any(nanmaskkernel))):
```
(python will short-circuit the if-statement; `np.any` seems more logical than counting).